torch/autograd/function.py

```diff
@@ -288,15 +288,22 @@ def _iter(obj):
 
 
 def _unflatten(input, proto):
+    from torch.nn.utils.rnn import PackedSequence
+
     # unflatten a list or tuple input into a nested list/tuple structure
     # specified by proto
     def unflatten_helper(input, proto):
         res = []
         if not isinstance(proto, (list, tuple)):
             return input[0], input[1:]
         for e in proto:
-            res_e, input = unflatten_helper(input, e)
-            res.append(res_e)
+            if e is None:
+                res.append(e)
+            else:
+                res_e, input = unflatten_helper(input, e)
+                res.append(res_e)
+        if isinstance(proto, PackedSequence):
```

Comment:
this is not a problem of this diff, but I wonder if there's a more elegant solution to handle this than importing random tuples.

One semi-hacky option is to spot any `namedtuple` instances. All of them seem to have `_make` method that takes iterable. So you can do `if hasattr(type(proto), '_make') return type(proto)(*res), input`

Then it'd work for any namedtuple and you'd avoid dependencies